src/python/pants/backend/python/lint/black/rules_integration_test.py

```diff
@@ -164,3 +179,33 @@ def test_skip(rule_runner: RuleRunner) -> None:
     assert not lint_results
     assert fmt_result.skipped is True
     assert fmt_result.did_change is False
+
+
+@skip_unless_python38_present
+def test_works_with_python38(rule_runner: RuleRunner) -> None:
+    """Black's typed-ast dependency does not understand Python 3.8, so we must instead run Black
+    with Python 3.8 when relevant."""
+    py38_sources = FileContent(
+        "py38.py",
+        dedent(
+            """\
+            import datetime
+
+            x = True
+            if y := x:
+                print("x is truthy and now assigned to y")
+
+
+            class Foo:
+                pass
```

Comment:
This was a real pain to get MyPy to error. If I just had the walrus operator portion, MyPy was fine..I don't know why we need the import and class declaration. I spent so much time trying to figure out how to get MyPy to error that I went with it once I got it working.